plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/actions/CodeModernizerStopModernizerAction.kt

```diff
@@ -25,12 +29,18 @@ class CodeModernizerStopModernizerAction :
     override fun update(event: AnActionEvent) {
         val project = event.project ?: return
         val codeModernizerManager = CodeModernizerManager.getInstance(project)
-        event.presentation.isEnabledAndVisible = codeModernizerManager.isModernizationJobActive()
+        event.presentation.isVisible = codeModernizerManager.isModernizationJobActive()
+        event.presentation.isEnabled = codeModernizerManager.isModernizationJobActive() && !codeModernizerManager.isModernizationJobStopping()
     }
 
     override fun actionPerformed(event: AnActionEvent) {
         val project = event.project ?: return
-        CodeModernizerManager.getInstance(project).userInitiatedStopCodeModernization()
+
+        runInEdt {
+            val toolWindow = ToolWindowManager.getInstance(project).getToolWindow(AmazonQToolWindowFactory.WINDOW_ID)
+            toolWindow?.show()
+        }
```

Comment:
nit: I'd not move logic of what should happen out of the manager. I.e. why should the logic for which toolwindow to be shown be owned by the button? To my mental model it makes more sense to control the lifecycle of the plugin from the Manager.